src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs

```diff
@@ -129,6 +130,18 @@ private void Write(ProjectEvaluationFinishedEventArgs e)
             Write(BinaryLogRecordKind.ProjectEvaluationFinished);
             WriteBuildEventArgsFields(e);
             Write(e.ProjectFile);
+
+            Write(e.ProfilerResult.HasValue);
+            if (e.ProfilerResult.HasValue && e.ProfilerResult.Value.ProfiledLocations.Count > 0)
```

Comment:
>e.ProfilerResult.Value.ProfiledLocations.Count > 0 [](start = 45, length = 50)

if it HasValue, but the Count is zero, this will confuse the reader. The condition here must match what we're writing in the line above.